var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -41,7 +41,7 @@ class Gptune(CMakePackage):
     depends_on("py-scikit-learn", type=("build", "run"))
     depends_on("py-matplotlib", type=("build", "run"))
     depends_on("py-pyyaml", type=("build", "run"))
-    depends_on("py-scikit-optimize@master+gptune", type=("build", "run"))
+    depends_on("py-scikit-optimize@0.5.2", patches=[patch("space.patch")], type=("build", "run"))
```

Comment:
I think several recent (github) releases do not work with this patch as the space.py file has changed a lot over the years. I would prefer to stick with scikit-optimize 0.5.2 for gptune.

Reply:
Gotcha. Unfortunately I don't know a way to specify a different patch depending on which version of py-scikit-optimize is chosen. Since we're hacking the internal library it's probably best to just pin it like you did here. I wonder if it's possible to submit this patch upstream so patching isn't required in the future.

Reply:
I'll think about it.

Reply:
> I think several recent (github) releases do not work with this patch

We're building `py-scikit-optimize@master+gptune` aka 0.9.0 in CI. The patch applies to 0.9.0, it does  not to 0.5.2.

Reply:
Also in your build instructions you're using `master`: https://github.com/gptune/GPTune#install-scikit-optimize